src/main/java/org/apache/commons/lang3/EnumUtils.java

```diff
@@ -294,6 +295,25 @@
         return map;
     }
 
+    /**
+     * <p>Gets the {@code Map} of enums.</p>
+     *
+     * <p>This method is useful when you need a map of enums.</p>
+     *
+     * @param <E> the type of enumeration
+     * @param <K> the type of the map key
+     * @param enumClass the class of the enum to query, not null
+     * @param keyFunction the function to query for the key, not null
+     * @return the modifiable map of enums, never null
+     */
+    public static <E extends Enum<E>, K> Map<K, E> getEnumMap(final Class<E> enumClass, final Function<E, K> keyFunction) {
+        final Map<K, E> map = new LinkedHashMap<>();
```

Comment:
Add `Objects.requireNonNull(enumClass, "enumClass")` as the 1st line of this method.

Reply:
Perhaps optimise the size of the hashmap:
```java
final E[] elements = enumClass.getEnumConstants();
final Map<K, E> map = new LinkedHashMap<>(elements.length);
for (final E e : elements) {
```